Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1497,14 +1501,23 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
-    // Attempt to shut down the tunnel
-    // On iOS 16 and below, as a workaround, we rekey to force a 403 error so that the tunnel fails to restart
     @MainActor
     private func attemptShutdown() async {
+        let cancelTunnel = {
+#if os(macOS)
+            try? self.tokenStore.deleteToken()
+#endif
+            self.cancelTunnelWithError(TunnelError.vpnAccessRevoked)
+        }
```

Comment:
Instead of rekeying we do this.  When starting up again the tunnel will rekey on its own if it still has an auth token.